lsp/nls/src/linearization/mod.rs

```diff
@@ -93,7 +93,7 @@ impl Default for AnalysisHost {
 impl Linearizer for AnalysisHost {
     type Building = Building;
     type Completed = Completed;
-    type CompletionExtra = (UnifTable, HashMap<usize, Ident>);
+    type CompletionExtra = (UnifTable, HashMap<usize, Ident>, Vec<TypeWrapper>);
```

Comment:
Nitpick: it may be time to turn this into a struct, just to have explicit names to describe what each component represents.

Reply:
Yeah, that makes sense. Since this struct would be a subset of the main type-checking `State` type, would it be better to just make the `CompletionExtra` associated type be the whole type-checking state, as in: 
```rust
type CompletionExtra = State
```

Reply:
I don't think that will work so well, because `State` is tied to a lifetime and retains mutables references and not owned values. I fear you'll have lifetime annotations that bubble up everywhere.